snuba/datasets/cdc/groupassignee.py

```diff
@@ -0,0 +1,72 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import ColumnSet, DateTime, Nullable, UInt
+from snuba.datasets.dataset_schemas import DatasetSchemas
+from snuba.datasets.cdc import CdcDataset
+from snuba.datasets.cdc.groupassignee_processor import GroupAssigneeProcessor, GroupAssigneeRow
+from snuba.datasets.schema import ReplacingMergeTreeSchema
+from snuba.snapshots.bulk_load import SingleTableBulkLoader
+
+
+class GroupAssigneeDataset(CdcDataset):
+    """
+    This is a clone of sentry_groupasignee table in postgres.
+
+    REMARK: the name in Clickhouse fixes the typo we have in postgres.
+    Since the table does not correspond 1:1 to the postgres one anyway
+    there is no issue in fixing the name.
+    """
+
+    POSTGRES_TABLE = "sentry_groupasignee"
+
+    def __init__(self) -> None:
+        columns = ColumnSet([
+            # columns to maintain the dataset
+            # Kafka topic offset
+            ("offset", UInt(64)),
+            ("record_deleted", UInt(8)),
+            # PG columns
+            ("id", UInt(64)),
+            ("project_id", UInt(64)),
+            ("group_id", UInt(64)),
+            ("date_added", Nullable(DateTime())),
+            ("user_id", Nullable(UInt(64))),
+            ("team_id", Nullable(UInt(64))),
+        ])
+
+        schema = ReplacingMergeTreeSchema(
+            columns=columns,
+            local_table_name='groupassignee_local',
+            dist_table_name='groupassignee_dist',
+            # TODO: add project id and group id to the identity key in postgres
+            # and then add them here.
+            order_by='(id)',
```

Comment:
I would make this (project_id, group_id, id). What do you think ?

Reply:
Makes sense to me. Since this is a unique index, this could even be `(project_id, group_id)` and we could drop `id` entirely if you want to.

Reply:
Is it unique per project_id+group_id? I tried to gather evidence of this on Sentry but did not manage to say that conclusively. There, the primary key is the id alone. We can put a uniquer index on project/group to enforce it though.